### ISSUE TITLE
[Bug] After displaying a page that contains a WebView, the FontSize of some controls (e.g. Label, Button) gets bigger

### ISSUE THREAD

**Issue by CrazyPhilip** [xamarin/Xamarin.Forms#10562]
### Description
When the app starts, all initialized pages do not contain any WebView, and it displays well. But after openning any page with a WebView, the FontSize of some controls, such as Label with <Label.FormattedText> or Button, becomes bigger. And I found that it had nothing to do with the Xamarin Forms version.
### Steps to Reproduce

1. Start the app, and all initialized pages DO NOT contain any WebView. It displays well.
2. Open a page without any WebView, and it displays well too.
3. Then return back to open another page with a WebView 
4. Return back to the previous page, some controls get bigger if the page has been re-rendered
5. Open the page in Step 2, some controls may change too.

### Expected Behavior
The FontSize of all controls should not change
### Actual Behavior
It changed.
### Basic Information

- Version with issue: 1.0
- Last known good version: no
- IDE: VS Professional 2019
- Platform Target Frameworks: 
  - iOS:  
  - Android: 9.0
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
```
<PackageReference Include="Xam.Plugin.TabView" Version="1.0.4" />
<PackageReference Include="Xam.TabView" Version="1.0.4" />
<PackageReference Include="Xamarin.Forms" Version="4.6.0.726" />
<PackageReference Include="Xamarin.Essentials" Version="1.5.3.1" />
```
- Affected Devices:

### Screenshots
![1](https://user-images.githubusercontent.com/30176398/80977852-e5632380-8e57-11ea-850e-49a60902f0c0.png)
![2](https://user-images.githubusercontent.com/30176398/80977896-f1e77c00-8e57-11ea-8116-315ba22df2b7.png)
![3](https://user-images.githubusercontent.com/30176398/80977898-f2801280-8e57-11ea-8e7a-6a3941ff9196.png)
![4](https://user-images.githubusercontent.com/30176398/80977904-f4e26c80-8e57-11ea-9058-3f9269c42b1d.png)

**Comment by rmarinho** [comment]
I @CrazyPhilip i try to reproduce but wasn't able, can you please share a small reproduction?! 
Does this happen only on Android 9 ?

**Comment by CrazyPhilip** [comment]
I test it on Android 9 and 10, and it happens on many devices. This is the repo: [https://github.com/CrazyPhilip/WebviewTest](url).

**Comment by jsuarezruiz** [comment]
@CrazyPhilip Thanks so much for the reproduction sample.
I have done several tests in a couple of emulators and cannot reproduce the issue. 
![issue10562](https://user-images.githubusercontent.com/6755973/83408721-67bd1400-a413-11ea-9eb8-467e5dbe788a.gif)

Does it happen to you in a specific device?.

**Comment by CrazyPhilip** [comment]
Thank you for your attention. 
The application that I'm actually developing used to use the native WebView, and our testers have all reported this issue, and the image (two screenshots overlap) below shows that the FontSize of the text has indeed changed. Just now I tested it on a virtual machine that was Pixel_2_pie_9_0_-_api_28, and it's the same result.
I've spent a little time trying to fix it, but there's really no way. So to get the project moving forward, I tried to use Tencent's X5 kernel to solve the problem, and it worked fine. 
![Screenshot_20200603_173611_com companyname2](https://user-images.githubusercontent.com/30176398/83621730-a395dc00-a5c1-11ea-9aa6-bd73e2fba393.png)

**Comment by jsuarezruiz** [comment]
I would like to continue testing this issue, from the screenshot that you have shared, could you indicate which device or emulator has been used?

**Comment by CrazyPhilip** [comment]
So far, I have tested it on HONOR V10, REDMI 8A, Smartisan Pro 2S, HUAWEI P10, a virtual machine Pixel_2_pie_9_0_-_api_28 and others' devices. And I noticed that it happened mainly on <Button> and <Label> with <span>. So I wonder if the WebView is affecting the rendering of these controls.

**Comment by CrazyPhilip** [comment]
We all use mobile phones that are common in mainland China, but I don't know if it has anything to do with Chinese-branded phones. 
I've simplified the code, and this example shows that WebView does affect controls like Label and Button. This is the repo: https://github.com/CrazyPhilip/WebviewTest.git
And please notice that ScaledDensity and Density are set to 2 in MainActivity.cs.

`Resources.DisplayMetrics.ScaledDensity = 2;`
 `Resources.DisplayMetrics.Density = 2;`

Steps To test the example:
1. Click the Button "1 New Page", and you will see NewPage pushed. (Left)
2. Go back and click the Button "2 Open Web", and a page with WebView will be pushed.
3. Go back and click the Button "1 New Page", and the same page in Step 1 pushed , but those controls might be changed. (Right)

**Only after a page with WebView is instantiated, does a new instantiated or refreshed page change.

Here are the Screenshots.
![screenshot](https://user-images.githubusercontent.com/30176398/84354447-7960a180-abf3-11ea-95c7-88ac5eef0c04.png)

**Comment by hartez** [comment]
> I don't know if it has anything to do with Chinese-branded phones.

I'm not suggesting it has anything to do with the brands of the phones. When I run your repro project on my devices and simulators, I'm not seeing the same things on the screen that you are - the sizes remain constant. 

I'm curious whether this might be a bug that shows up only for certain language/region settings.

**Comment by hartez** [comment]
Aha! I _can_ reproduce it if I use a higher density display device. On a device that starts with a ScaledDensity of 3.5, the size of the items on the first page will change after displaying the WebView. On devices with a density closer to 2, there's no change.

**Comment by hartez** [comment]
And if I comment out the lines where you are manually setting the ScaledDensity and the Density, the item sizes remain constant.

@CrazyPhilip Why are you adjusting the Density and ScaledDensity in OnCreate?

**Comment by CrazyPhilip** [comment]
Actually I came up with this idea in the first place to accommodate devices with different resolutions. So then the problem came up.
Maybe I was wrong. I don't know if this is a bug or not now. 🤣 
@hartez

**Comment by CrazyPhilip** [comment]
Android allow users to adjust the Text size and Display size manually, but my app would run on devices with different settings and it would be affected by the settings. So I want to manually set the ScaledDensity and the Density in OnCreate to make the items size remain constant.

**Comment by hartez** [comment]
> Android allow users to adjust the Text size and Display size manually, but my app would run on devices with different settings and it would be affected by the settings. So I want to manually set the ScaledDensity and the Density in OnCreate to make the items size remain constant.

Generally speaking, this is not a good idea. Users might be adjusting these settings for accessibility reasons; changing them in your app might be making it difficult for them to use it. 

I would suggest leaving the Density and ScaledDensity alone and adapting your app design to accommodate different densities.

**Comment by CrazyPhilip** [comment]
Yeah sure, so in the newest app it has been changed to accommodate devices automatically. 
Maybe this is just a misunderstanding, but thank you for your answers. 😃
